coalib/output/ConsoleInteraction.py

```diff
@@ -135,32 +135,37 @@ def acquire_actions_and_apply(console_printer,
     """
     cli_actions = CLI_ACTIONS if cli_actions is None else cli_actions
     failed_actions = set()
-    while True:
-        actions = []
-        for action in cli_actions:
-            if action.is_applicable(result, file_dict, file_diff_dict) is True:
-                actions.append(action)
-
-        if actions == []:
-            return
+    successful_actions = []
 
+    while True:
         action_dict = {}
         metadata_list = []
-        for action in actions:
-            metadata = action.get_metadata()
-            action_dict[metadata.name] = action
-            metadata_list.append(metadata)
+
+        for action in cli_actions:
+            if action.is_applicable(result,
+                                    file_dict,
+                                    file_diff_dict,
+                                    successful_actions) is True:
+                metadata = action.get_metadata()
+                action_dict[metadata.name] = action
+                metadata_list.append(metadata)
+
+        if not metadata_list:
+            return
 
         # User can always choose no action which is guaranteed to succeed
-        if not ask_for_action_and_apply(console_printer,
-                                        section,
-                                        metadata_list,
-                                        action_dict,
-                                        failed_actions,
-                                        result,
-                                        file_diff_dict,
-                                        file_dict):
+        chosen_action = ask_for_action_and_apply(console_printer,
+                                                 section,
+                                                 metadata_list,
+                                                 action_dict,
+                                                 failed_actions,
+                                                 result,
+                                                 file_diff_dict,
+                                                 file_dict)
+        if not chosen_action:
             break
+        else:
+            successful_actions.append(chosen_action)
```

Comment:
Yes, all action (names) that are chosen by the user are stored there. So if an error happens they will be included too.
I can rename the variable, or I can try to handle it differently, i.e. failed actions are not included in successful actions, which would you prefer?

I just noticed that I didn't actually change the return type of `ask_for_action_and_apply` from action_name to the action, which confuses me, because I thought I had. 
I will do that now :)